nengo_viz/static/viz_slider.js

```
@@ -212,12 +226,105 @@ VIZ.Slider.prototype.generate_menu = function() {
+    if (button == 13) {
+        var self = this;
+        var slider_range = self.scale.domain();
+        var msg = document.getElementById('value_in_field').value;
+        $(text_div).off('keypress');
+        if (VIZ.is_num(msg)) {
+            self.set_value(ind, VIZ.max_min(Number(msg), slider_range[1], slider_range[0]));
+            return;
+        }
+        else {
+            alert('failed to set value');
+            text_div.innerHTML = 0;
+            self.set_value(ind, VIZ.max_min(0, slider_range[1], slider_range[0]));   
+            return;
+        }
+    }
+};
+
+/** report an event back to the server */
+VIZ.Slider.prototype.notify = function(info) {
+    this.notify_msgs.push(info);
+    
+    // only send one message at a time
+    // TODO: find a better way to figure out when it's safe to send
+    // another message, rather than just waiting 1ms....
+    if (this.notify_msgs.length == 1) {
+        var self = this;
+        window.setTimeout(function() {
+            self.send_notify_msg();
+        }, 1);
+    }
+}
+
+/** send exactly one message back to server
+ *  and schedule the next message to be sent, if any
+ */
+VIZ.Slider.prototype.send_notify_msg = function() {
+    msg = this.notify_msgs[0];
+    this.ws.send(msg);
+    if (this.notify_msgs.length > 1) {
+        var self = this;
+        window.setTimeout(function() {
+            self.send_notify_msg();
+        }, 1);
+    }
+    this.notify_msgs.splice(0, 1);
+}
+
+VIZ.Slider.prototype.user_value = function () {
+    var self = this;
+    var new_value = prompt('Set value\nExample: 1.3, 4.2');
+    
+    if (new_value == null) {
+        return;
+    };
+    new_value = new_value.split(',');
+    var slider_range = this.scale.domain();
+
+    for (var i = 0; i < this.sliders.length; i++){
+        if (!(VIZ.is_num(new_value[i]))) {
+            alert("invalid input :" + new_value[i] + "\nFor the slider in position " + (i +1) );
```

I'm very much not fond of popping up a whole bunch of alert messages. In the old visualizer, we just failed silently (i.e. no alert message at all).